numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -507,6 +507,77 @@ string_expandtabs_loop(PyArrayMethod_Context *context,
 }
 
 
+template <ENCODING bufferenc, ENCODING fillenc>
+static int
+string_center_ljust_rjust_loop(PyArrayMethod_Context *context,
+        char *const data[], npy_intp const dimensions[],
+        npy_intp const strides[], NpyAuxData *NPY_UNUSED(auxdata))
+{
+    JUSTPOSITION pos = *(JUSTPOSITION *)(context->method->static_data);
+    int elsize1 = context->descriptors[0]->elsize;
+    int elsize3 = context->descriptors[2]->elsize;
+    int outsize = context->descriptors[3]->elsize;
+
+    char *in1 = data[0];
+    char *in2 = data[1];
+    char *in3 = data[2];
+    char *out = data[3];
+
+    npy_intp N = dimensions[0];
+
+    while (N--) {
+        Buffer<bufferenc> buf(in1, elsize1);
+        Buffer<fillenc> fill(in3, elsize3);
+        Buffer<bufferenc> outbuf(out, outsize);
+        size_t len = string_pad(buf, *(npy_int64 *)in2, *fill, pos, outbuf);
```

Comment:
this should be `npy_intp len`, otherwise the error check below will never succeed.

Reply:
also what happens if `out` and `in1` are the same? Do you need to allocate a temporary buffer to avoid clobbering the input in that case like in my implementation?